pkg/vault/client.go

```diff
@@ -438,6 +449,19 @@ func GetGroupInfo(instanceAddr string, name string) (map[string]interface{}, err
 	return entity.Data, nil
 }
 
+// write secret to vault
+func WriteApproleSecretID(instanceAddr, secretPath string) (*api.Secret, error) {
+	secret, err := getClient(instanceAddr).Logical().Write(secretPath, map[string]interface{}{})
```

Comment:
what data do we write here? an empty map ?

Reply:
or is this actually retrieving the role secret data in order to store it later ? It just looks weird we need to use `Write`...

Reply:
The `write` was a strange concept for me as well. This is based upon https://www.vaultproject.io/docs/auth/approle#configuration , specifically the `vault write -f auth/approle/role/my-role/secret-id` that is required to generate a secret id for a particular role. I will update the comments around this function to better elaborate.

Reply:
The [api approach](https://www.vaultproject.io/docs/auth/approle#via-the-api-1) may be more clear. They're making a `POST` to the secret-id endpoint with an empty body to generate a new `secret_id`. However, the `Write()` function of the Go vault package requires a data payload parameter, so an empty map is passed.

Reply:
understood, thanks for the explanation. Would it be worth renaming the `WriteApproleSecretID` method to something clearer, like `GenerateApproleSecret` or something like that ?

Reply:
Great point. Naming it after what's technically occurring is not very helpful in this case :grimacing: